test/functional/test_framework/mininode.py

```diff
@@ -534,7 +534,7 @@ def send_blocks_and_test(self, blocks, node, *, success=True, force_send=False,
                 for b in blocks:
                     self.send_message(msg_block(block=b))
             else:
-                self.send_message(msg_headers([CBlockHeader(blocks[-1])]))
+                self.send_message(msg_headers([CBlockHeader(block) for block in blocks]))
```

Comment:
I think we want to send only the latest header and wait for the other side to ask for all on its own.

Reply:
That's currently considered a network protocol error...?

Reply:
Eh, yeah